### PR TITLE
replace obsolete pycrypto with pycryptodome in Python 3.10.x easyconfigs

### PR DESCRIPTION
**Pull request by surak**
(created using `eb --new-pr`)

These easyconfig files come from "develop" , so there are a couple other changes, apparently, which are not on the current tag. 

This is a fix for 
https://github.com/easybuilders/easybuild-easyconfigs/issues/17410

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen2
CORE_CNT=16
EB_ARGS="--installpath /tmp/$USER/pr17412"

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso
CORE_CNT=16
EB_ARGS="--installpath /tmp/$USER/pr17412"

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bwd-rockylinux-91 - Linux Rocky Linux 9.1 (Blue Onyx), x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 525.60.13, Python 3.9.14
See https://gist.github.com/d8d1af9692654248d00668cebdac2a2b for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
skl-rockylinux-87 - Linux Rocky Linux 8.7, x86_64, Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz (skylake), 1 x NVIDIA NVIDIA RTX A4000, 525.85.12, Python 3.6.8
See https://gist.github.com/07a8626e79b04a3a70057afca9b09dad for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
skl-opensuse-154 - Linux openSUSE Leap 15.4, x86_64, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (skylake), Python 3.6.15
See https://gist.github.com/0fb446e254e40524e98a2b3477a4fc37 for a full test report.

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
zen2-ubuntu-2210 - Linux Ubuntu 22.10 (Kinetic Kudu), x86_64, AMD EPYC 7452 32-Core Processor (zen2), Python 3.10.7
See https://gist.github.com/fd1463ef72c9df3c05714ff4d17ec2b0 for a full test report.

File: easybuild/easyconfigs/p/Python/Python-3.10.4-GCCcore-11.3.0.eb [hunk 1]
```diff
@@ -99,14 +99,9 @@ exts_list = [
         'modulename': 'arff',
         'checksums': ['3220d0af6487c5aa71b47579be7ad1d94f3849ff1e224af3bf05ad49a0b5c4da'],
     }),
-    ('pycrypto', '2.6.1', {
+    ('pycryptodome', '3.17', {
         'modulename': 'Crypto',
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-3.10.4-GCCcore-11.3.0.eb:103]: We should also use `Crypto.PublicKey.RSA` here to catch the broken `pycrypto`?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/p/Python/Python-3.10.4-GCCcore-11.3.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/p/Python/Python-3.10.4-GCCcore-11.3.0.eb]: fixed in 545a571c1a

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--skip"

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen2
EB_ARGS="--skip"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3121.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/5193cd10c54dd40d3470d705d691196a for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 6 out of 6 (2 easyconfigs in total)
vera-c2 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/dbe9ed8bfe4a98c15703620c40284583 for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @surak!

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen2

**Pull request merged**